docs/extensibility/payment-method-integration.md

```diff
@@ -63,7 +63,7 @@ Here's some more details on the configuration options:
 -   `name` (required): This should be a unique string (wise to try to pick something unique for your gateway that wouldn't be used by another implementation) that is used as the identifier for the gateway client side. If `paymentMethodId` is not provided, `name` is used for `paymentMethodId` as well.
 -   `content` (required): This should be a react node that will output in the express payment method area when the block is rendered in the frontend. It will be cloned in the rendering process. When cloned, this react node will receive props from the payment method interface to checkout that will allow your component to interact with checkout data (more on this later).
 -   `edit` (required): This should be a react node that will be output in the express payment method area when the block is rendered in the editor. It will be cloned in the rendering process. When cloned, this react node will receive props from the payment method interface to checkout (but they will contain preview data).
--   `canMakePayment` (required): This should be a function that returns whether the payment method can make a payment in the current environment. The function _can_ return a promise and will receive [`cartData`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/308e968c700028180cab391f2223eb0a43dd2d4d/assets/js/type-defs/cart.js#L163-L177) as an argument (so the payment method can use that to determine if it can be used). Most payment methods will probably just return `true` from the function, but this does allow payment methods like `ApplePay` which have restrictions on where they work to return `false` and thus be excluded as an option in specific browser environments (or other conditions). Keep in mind this function is only invoked once in the mounting process of the checkout block.
+-   `canMakePayment` (required): A callback to determine whether the payment method should be available as an option for the shopper. The function will be passed an object containing data about the current order. Return a boolean value - true if payment method is available for use. If your gateway needs to perform async initialisation to determine availability, you can return a promise (resolving to boolean). This allows a payment method to be hidden based on the cart, e.g. if the cart has physical/shippable products (example: `Cash on delivery`). The function Keep in mind this function is only invoked once in the mounting process of the checkout block. Also can be used to disable a payment method based on environment, browser or geolocation restrictions (example: `Apple Pay`). 
```

Comment:
We could make this optional - default to `true`. Also, could also support payment methods just returning boolean directly; i.e. detect if it's a callback or promise and handle accordingly.

Reply:
> Also, could also support payment methods just returning boolean directly; i.e. detect if it's a callback or promise and handle accordingly.

This already happens:

> https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/0421af44603090cebe9df79cee04c62bbf1959c1/assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js#L91-L93

The value returned is wrapped in a Promise and resolved immediately. So it handles whether it's a promise or not already. The only thing it doesn't handle is if it's a straight boolean or not which we can add.

Reply:
> We could make this optional - default to true. 

I'm on the fence with this. I think it's useful to signal to payment methods that this property should be provided. I realize there are a few payment methods that will always be `true` here.

Reply:
> The only thing it doesn't handle is if it's a straight boolean or not which we can add.

Sounds good - I guess we can do that as follow up? Not particularly important for this PR. If we allow plain boolean then there's no urgent need for a default (trivial methods just pass `true`).

Reply:
> Sounds good - I guess we can do that as follow up?

We could, I'm not sure whether it needs to be high priority though (or even done this cycle). I see it more as a "nice to have" than "must have".